### PR TITLE
Gd/integration tests

### PR DESCRIPTION
**Pull request by guipublic**
Add integration tests for nargo that try to prove and then verify noir programs located in nargo/test_data.
An optional config.toml file in test_data can exclude programs to be be tested or they can be marked as fail meaning the verification of their proof should fail.

### CODE REVIEW

File: crates/nargo/tests/test_data/2_div/src/foo.nr [hunk 1]
```diff
@@ -0,0 +1,3 @@
+fn hello(x : pub Field) -> Field {
```
Comment (kevaundray) [inline crates/nargo/tests/test_data/2_div/src/foo.nr:1]: This file is not being used, so it should be removed. Can you do it for all of the examples?

File: crates/nargo/tests/test_data/2_div/src/main.nr [hunk 2]
```diff
@@ -0,0 +1,16 @@
+mod foo;
```
Comment (kevaundray) [inline crates/nargo/tests/test_data/2_div/src/main.nr:1]: Lets delete the multiline comment from this file. In the future, we should include comments which say what it actually happening and what we are testing. Can you do this for all of the examples?

File: crates/nargo/tests/test_data/4_sub/Prover.toml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#x = "10"
```
Comment (kevaundray) [inline crates/nargo/tests/test_data/4_sub/Prover.toml:1]: Do we need these commented out numbers?

File: crates/nargo/tests/test_data/5_over/Prover.toml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+x = "43046721"
+y = "3793632897"
+#z = "2418266113"    #x10
```
Comment (kevaundray) [inline crates/nargo/tests/test_data/5_over/Prover.toml:3]: Same here for these commented out numbers

File: crates/nargo/tests/test_data/5_over/src/main.nr [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+// Release : 0.2
+//
+// To run a proof on the command line, type `cargo run prove {proof_name}`
+//
+// To verify that proof, type `cargo run verify {proof_name}`
+
+fn main(x : u32, y : u32) {
+    x = x * x;
+    constrain (y) == x;
+    // x = x * x;
```
Comment (kevaundray) [inline crates/nargo/tests/test_data/5_over/src/main.nr:16]: Can you remove the commented out code?

File: crates/nargo/src/cli/verify_cmd.rs [hunk 6]
```diff
@@ -112,3 +83,38 @@ pub fn add_dummy_setpub_arr(abi: &mut Abi) {
     // ensuring there will be no conflicts
     abi.parameters.push((RESERVED_PUBLIC_ARR.into(), dummy_arr));
 }
+
+pub fn verify_hlp(
+    prg_dir: &Path, //&std::path::PathBuf,
+    proof_path: std::path::PathBuf,
```
Comment (kevaundray) [inline crates/nargo/src/cli/verify_cmd.rs:89]: It doesn't seem like proof_path or prg_dir are changed here so maybe we can use `AsRef` something like `fn foo<P: AsRef<Path>>(p: P)` ?

- Nit: Can you rename prg_dir to program_dir

File: crates/nargo/src/cli/verify_cmd.rs [hunk 7]
```diff
@@ -112,3 +83,38 @@ pub fn add_dummy_setpub_arr(abi: &mut Abi) {
     // ensuring there will be no conflicts
     abi.parameters.push((RESERVED_PUBLIC_ARR.into(), dummy_arr));
 }
+
+pub fn verify_hlp(
```
Comment (kevaundray) [inline crates/nargo/src/cli/verify_cmd.rs:87]: Lets rename verify_hlp to `verify_with_path`

File: crates/nargo/src/cli/prove_cmd.rs [hunk 8]
```diff
@@ -119,3 +84,53 @@ fn process_abi_with_input(
     }
     Ok(solved_witness)
 }
+
+pub fn prove_hlp(
```
Comment (kevaundray) [inline crates/nargo/src/cli/prove_cmd.rs:88]: Lets rename prove_hlp to `prove_with_path`

- Nit: can you rename prg_dir to program_dir?

Comment (kevaundray) [reply crates/nargo/src/cli/prove_cmd.rs:88]: Also same suggestion about using AsRef<Path> , PathBuf ideally should be used when we are going to modify the path again

**Comment by kevaundray** [comment]
Looks good modulo the comments. Thanks!

File: crates/nargo/tests/test_data/2_div/src/foo.nr (hunk 1 above)
Comment (guipublic) [reply crates/nargo/tests/test_data/2_div/src/foo.nr:1]: yes, it's done.

File: crates/nargo/tests/test_data/2_div/src/main.nr (hunk 2 above)
Comment (guipublic) [reply crates/nargo/tests/test_data/2_div/src/main.nr:1]: yes, it's done

File: crates/nargo/tests/test_data/5_over/src/main.nr (hunk 5 above)
Comment (guipublic) [reply crates/nargo/tests/test_data/5_over/src/main.nr:16]: Done

File: crates/nargo/src/cli/verify_cmd.rs (hunk 7 above)
Comment (guipublic) [reply crates/nargo/src/cli/verify_cmd.rs:87]: Done

File: crates/nargo/tests/test_data/5_over/Prover.toml (hunk 4 above)
Comment (guipublic) [reply crates/nargo/tests/test_data/5_over/Prover.toml:3]: Done

File: crates/nargo/tests/test_data/2_div/src/foo.nr (hunk 1 above)
Comment (guipublic) [reply crates/nargo/tests/test_data/2_div/src/foo.nr:1]: Yes, it's done.

**Comment by kevaundray** [comment]
LGTM

**Pull request merged**